### PR TITLE
[DOCS-6126] Digital Workspace developer documentation

### PR DESCRIPTION
**Pull request by oliveralfresco2018**

### CODE REVIEW

File: digital-workspace/latest/develop/extension-mechanism.md [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+### Create a Digital Workspace extension
+
+Use the [Nx](https://nx.dev/) developer tools for monorepos to create the Digital Workspace extension.
+
+1. Install `nx` cli globally.
+
+    `npm install -g nx`
+
+2. Use the following command to create a new extension called `my-extension`, from the root directory of the Digital Workspace project.
+
```
Comment (aborroy) [inline digital-workspace/latest/develop/extension-mechanism.md:34]: It should be required to add a link to document section describing where to obtain and configure a Digital Workspace project.

File: digital-workspace/latest/develop/extension-mechanism.md [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+            ...,
+            MyExtensionModule,
+        ],
+    })
+    export class AppExtensionsModule {}
+    ```
+
+7. Edit the `angular.json` configuration file so the extension is visible from the Digital Workspace app.
+
+8. To the `projects/content-ee/targets/build/options/assets` file add the following.
```
Comment (aborroy) [inline digital-workspace/latest/develop/extension-mechanism.md:113]: This sentence is a continuation from the last one. You need to modify the `angular.json` file adding the following lines to the section `projects/content-ee/targets/build/options/assets` (this is a section inside the angular.json file)

**Review by aborroy — CHANGES_REQUESTED** [review]
Some minor corrections have been added

File: digital-workspace/latest/develop/extension-mechanism.md [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+
+### Create a Digital Workspace extension
+
+Use the [Nx](https://nx.dev/) Workspace developer tools for monorepos to create the Digital Workspace extension.
+
+1. Install `nx` cli globally.
+
+  `npm install -g nx`
+
+2. Use the following command to create a new extension called `my-extension`, from the root directory of the Digital Workspace project.
```
[the remark this answers is not in the document]
Comment (oliveralfresco2018) [reply digital-workspace/latest/develop/extension-mechanism.md:33]: Added link to the small pre req section above. 
cheers Angel

File: digital-workspace/latest/develop/extension-mechanism.md (hunk 2 above)
Comment (oliveralfresco2018) [reply digital-workspace/latest/develop/extension-mechanism.md:113]: Added the two together.

File: digital-workspace/latest/develop/extension-mechanism.md (hunk 1 above)
Comment (oliveralfresco2018) [reply digital-workspace/latest/develop/extension-mechanism.md:34]: See above

**Review by aborroy — DISMISSED** [review]
LGTM

**Review by gravitonian — CHANGES_REQUESTED** [review]
Shouldn't http://localhost:4000/digital-workspace/latest/extensions/ be merged into the http://localhost:4000/digital-workspace/latest/develop/ section somehow?

I think the "Develop/Develop Digital Workspace" section should be called "Develop/Deploy the application" - it does not tell you how to develop/customize anything.

The "Develop/Manage the Digital Workspace" is 50% a copy of "Develop/Develop Digital Workspace". I would suggest renaming it to "Develop/Customizing the application" and show some examples of customizations/extensions. Move the "Build, promote, test" section to the deploy section.

The "Develop/Extension mechanism" should be part of a "Develop/Customizing the application" section.

Difficult to review this at this point...

**Comment by anxumalo** [comment]
@gravitonian I've now applied some of the restructuring that we discussed - see what you think. Thanks

**Review by anxumalo — APPROVED** [review]

**Review by anxumalo — APPROVED** [review]
Merging updates so far so the content is published. More refinement is needed - to be tracked in a new ticket.

**Pull request merged**